src/uikit.cs

```diff
@@ -16610,6 +16754,505 @@ interface UITargetedDragPreview : NSCopying
+
+		[Export ("tableView:dragSessionWillBegin:")]
+		void DragSessionWillBegin (UITableView tableView, IUIDragSession session);
+
+		[Export ("tableView:dragSessionDidEnd:")]
+		void DragSessionDidEnd (UITableView tableView, IUIDragSession session);
+
+		[Export ("tableView:dragSessionAllowsMoveOperation:")]
+		bool DragSessionAllowsMoveOperation (UITableView tableView, IUIDragSession session);
+
+		[Export ("tableView:dragSessionIsRestrictedToDraggingApplication:")]
+		bool DragSessionIsRestrictedToDraggingApplication (UITableView tableView, IUIDragSession session);
+	}
+
+	[NoWatch, NoTV]
+	[iOS (11,0)]
+	[Protocol, Model]
+	[BaseType (typeof(NSObject))]
+	interface UITableViewDropDelegate {
+		[Abstract]
+		[Export ("tableView:performDropWithCoordinator:")]
+		void PerformDrop (UITableView tableView, IUITableViewDropCoordinator coordinator);
+
+		[Export ("tableView:canHandleDropSession:")]
+		bool CanHandleDropSession (UITableView tableView, IUIDropSession session);
+
+		[Export ("tableView:dropSessionDidEnter:")]
+		void DropSessionDidEnter (UITableView tableView, IUIDropSession session);
+
+		[Export ("tableView:dropSessionDidUpdate:withDestinationIndexPath:")]
+		UITableViewDropProposal DropSessionDidUpdate (UITableView tableView, IUIDropSession session, [NullAllowed] NSIndexPath destinationIndexPath);
+
+		[Export ("tableView:dropSessionDidExit:")]
+		void DropSessionDidExit (UITableView tableView, IUIDropSession session);
+
+		[Export ("tableView:dropSessionDidEnd:")]
+		void DropSessionDidEnd (UITableView tableView, IUIDropSession session);
+
+		[Export ("tableView:dropPreviewParametersForRowAtIndexPath:")]
+		[return: NullAllowed]
+		UIDragPreviewParameters GetDropPreviewParameters (UITableView tableView, NSIndexPath indexPath);
+	}
+
+	[NoWatch, NoTV]
+	[iOS (11,0)]
+	[BaseType (typeof(UIDropProposal))]
+	[DisableDefaultCtor] // NSInternalInconsistencyException Reason: Not implemented
+	interface UITableViewDropProposal {
+
+		// inline from base type
+		[Export ("initWithDropOperation:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (UIDropOperation operation);
+
+		[Export ("initWithDropOperation:intent:")]
+		IntPtr Constructor (UIDropOperation operation, UITableViewDropIntent intent);
+
+		[Export ("intent")]
+		UITableViewDropIntent Intent { get; }
+	}
+
+	[NoWatch, NoTV]
+	[iOS (11,0)]
+	[Protocol]
+	interface UITableViewDropCoordinator {
+		[Abstract]
+		[Export ("items")]
+		IUITableViewDropItem[] Items { get; }
+
+		[Abstract]
+		[NullAllowed, Export ("destinationIndexPath")]
+		NSIndexPath DestinationIndexPath { get; }
+
+		[Abstract]
+		[Export ("proposal")]
+		UITableViewDropProposal Proposal { get; }
+
+		[Abstract]
+		[Export ("session")]
+		IUIDropSession Session { get; }
+
+		[Abstract]
+		[Export ("dropItem:toPlaceholder:")]
+		IUITableViewDropPlaceholderContext DropItemToPlaceholder (UIDragItem dragItem, UITableViewDropPlaceholder placeholder);
```

Comment:
@sebastien mmh it’s kinda confusing in C# as I don’t think we should name the 2nd argument `toPlaceholder` right.

We'd get:

``` C#
IUICollectionViewDropPlaceholderContext Drop (UIDragItem dragItem, UICollectionViewDropPlaceholder placeholder);

IUIDragAnimating Drop (UIDragItem dragItem, NSIndexPath indexPath);

IUIDragAnimating Drop (UIDragItem dragItem, NSIndexPath indexPath, CGRect rect);

IUIDragAnimating Drop (UIDragItem dragItem, UIDragPreviewTarget target);
```

We're loosing context here when calling those methods compared to Swift where the `to` or `toItemAt` stays when calling the method:

``` Swift
drop(myDragItem, to: myPlaceholder)
drop(myDragItem, toItemAt: myIndexPath)
drop(myDragItem, intoItemAt: myIndexPath, rect: myRect)
drop(myDragItem, to: myTarget)
```

We'd have 4 `Drop` methods and I don't think the arguments types alone give enough context.

I think we can go `DropToPlaceholder`, `DropToItem`, etc, etc... though.

Do you agree?